### PR TITLE
Use debug! logging instead of info! for verbose logging

### PR DESCRIPTION
**Pull request by rylev**
This issue has not been discussed, but I believe it simple enough that a discussion wasn't needed.  The `info!` log is meant for high level logs that are relevant to every invocation. The logs in this PR are only relevant when debugging so they should use the `debug!` logging macro.

I'm not sure who should look at this, but perhaps @peterhuene can take a look.

### CODE REVIEW

File: crates/environ/src/cache/worker.rs [hunk 1]
```diff
@@ -100,7 +100,7 @@ impl Worker {
         let sent_event = self.sender.try_send(event.clone());
 
         if let Err(ref err) = sent_event {
-            info!(
+            debug!(
```
Comment (sunfishcode) [inline crates/environ/src/cache/worker.rs:103]: This error seems is about more than just "here's some stuff you may want if you're debugging". It's more "something unexpected happened which isn't immediately fatal but which may mean the module cache isn't actually caching things".

File: crates/jit/src/compiler.rs [hunk 2]
```diff
@@ -158,7 +158,7 @@ impl Compiler {
             // show up be sure to log it in case anyone's listening and there's
             // an accidental bug.
             if relocations.len() > 0 {
-                log::info!("relocations found in trampoline for {:?}", sig);
+                log::debug!("relocations found in trampoline for {:?}", sig);
```
Comment (sunfishcode) [inline crates/jit/src/compiler.rs:161]: This also seems higher priority than debug. This is "this should be super rare, and even though we have code to handle it, be aware that this super rare thing is happening!"

**Comment by rylev** [comment]
@sunfishcode no it doesn't seem like there's good docs on when to use which log. From what I've seen out there
* `info` is high-level logs that you'd want to show on every normal run of the program
* `warn` is something went wrong but it's not fatal and should just be checked out when you have a chance
* `error` an error occured. It's not enough to crash the program but it's something that should be looked into ASAP
*`debug` any info that would be relevant to see the flow of the executing program
* `trace` very verbose logs. Generally not helpful for program wide debugging but useful when inspecting a small segment of the program.

File: crates/jit/src/compiler.rs (hunk 2 above)
Comment (rylev) [reply crates/jit/src/compiler.rs:161]: This sounds like `warn` might be more appropriate then.

File: crates/environ/src/cache/worker.rs (hunk 1 above)
Comment (rylev) [reply crates/environ/src/cache/worker.rs:103]: this sounds like `warn` again

Comment (sunfishcode) [reply crates/environ/src/cache/worker.rs:103]: Makes sense. Would you mind changing this one to `warn`?

File: crates/jit/src/compiler.rs (hunk 2 above)
Comment (sunfishcode) [reply crates/jit/src/compiler.rs:161]: This message in itself doesn't indicate that something went wrong, just that something we expect to be rare, but still valid, happened. I'll go with your judgement about whether this should be `info` or `warn` though.

**Comment by bjorn3** [comment]
> It looks like CI failed to build because ld couldn't link, but it gave no discernible diagnostic message, just returned an exit code of 1. Hopefully just a transient failure.

That always happens for me when the disk is full.

**Comment by peterhuene** [comment]
The failure is that CI is running out of disk space and we have some PRs up that will hopefully help mitigate the problem shortly.

**Comment by rylev** [comment]
@peterhuene This should be good to go now. Sorry for the delay!

**Comment by cfallin** [comment]
@rylev I'm cleaning up old PRs and came upon this; are you still interested in this change? (Lots of conflicts now but it seems it wouldn't be too hard to redo the change in general) Sorry this got dropped on the floor beforehand!

**Comment by rylev** [comment]
@cfallin no worries! I'll close this. The user can always filter the logs fairly easily if it's too verbose.

**Pull request closed without merging**